src/renderer/redux/actions/search.js

```diff
@@ -24,107 +19,160 @@ export const doSearch = rawQuery => (dispatch, getState) => {
     return;
   }
 
+  // If we have already searched for something, we don't need to do anything
+  const urisForQuery = makeSelectSearchUris(query)(state);
+  if (urisForQuery && !!urisForQuery.length) {
+    return;
+  }
+
+  // If the user is on the file page with a pre-populated uri and they select
+  // the search option without typing anything, searchQuery will be empty
+  // We need to populate it so the input is filled on the search page
+  if (!state.search.searchQuery) {
+    dispatch({
+      type: ACTIONS.UPDATE_SEARCH_QUERY,
+      data: { searchQuery: query },
+    })
+  }
+
   dispatch({
-    type: ACTIONS.SEARCH_START,
-    data: { query },
+    type: ACTIONS.SEARCH_START
   });
 
-  if (page !== 'search') {
-    dispatch(doNavigate('search', { query }));
-  } else {
-    fetch(`https://lighthouse.lbry.io/search?s=${query}`)
-      .then(handleSearchApiResponse)
-      .then(data => {
-        const uris = [];
-        const actions = [];
-
-        data.forEach(result => {
-          const uri = buildURI({
-            name: result.name,
-            claimId: result.claimId,
-          });
-          actions.push(doResolveUri(uri));
-          uris.push(uri);
-        });
+  fetch(`https://lighthouse.lbry.io/search?s=${query}`)
+    .then(handleFetchResponse)
+    .then(data => {
+      const uris = [];
+      const actions = [];
 
-        actions.push({
-          type: ACTIONS.SEARCH_SUCCESS,
-          data: {
-            query,
-            uris,
-          },
-        });
-        dispatch(batchActions(...actions));
-      })
-      .catch(() => {
-        dispatch({
-          type: ACTIONS.SEARCH_FAIL,
+      data.forEach(result => {
+        const uri = buildURI({
+          name: result.name,
+          claimId: result.claimId,
         });
+        actions.push(doResolveUri(uri));
+        uris.push(uri);
       });
-  }
+
+      actions.push({
+        type: ACTIONS.SEARCH_SUCCESS,
+        data: {
+          query,
+          uris,
+        },
+      });
+      dispatch(batchActions(...actions));
+    })
+    .catch(() => {
+      dispatch({
+        type: ACTIONS.SEARCH_FAIL,
+      });
+    });
 };
 
-export const updateSearchQuery = searchQuery => ({
-  type: ACTIONS.UPDATE_SEARCH_QUERY,
-  data: { searchQuery },
-});
+export const doUpdateSearchQuery =
+  (query: string, shouldSkipSuggestions: ?boolean) => dispatch => {
+    dispatch({
+      type: ACTIONS.UPDATE_SEARCH_QUERY,
+      data: { query },
+    })
+
+    // Don't fetch new suggestions if the user just added a space
+    if (!query.endsWith(" ") || !shouldSkipSuggestions) {
+      dispatch(getSearchSuggestions(query));
+    }
+  };
+
+export const getSearchSuggestions = (value: string) => (dispatch, getState) => {
+  const { search: searchState } = getState();
+  const query = value.trim();
+
+  const isPrefix = () => {
```

Comment:
doesn't really matter, but maybe you should use regex here

`/^(@|lbry:\/{0,2})$/`